src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -4,11 +4,25 @@
 
 from __future__ import absolute_import, division, print_function
 
+import os
 import threading
 
 from cryptography.hazmat.bindings._openssl import ffi, lib
 
 
+@ffi.callback("int (*)(unsigned char *, int)", error=-1)
```

Comment:
Everysingle caller of `RAND_bytes` in `ssl/` uses `<= 0` to check the
status, so it seems to be immaterial.

On Sun, Jun 28, 2015 at 3:58 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/bindings/openssl/binding.py
> <https://github.com/pyca/cryptography/pull/2073#discussion_r33426295>:
>
> >  import threading
> >
> >  from cryptography.hazmat.bindings._openssl import ffi, lib
> >
> >
> > +@ffi.callback("int (*)(unsigned char *, int)", error=-1)
>
> -1 is definitely the right response for pseudo (since that's a terrible
> response code system and we handled it that way in the past), but I'm not
> convinced it's the right one for RAND_bytes. It's *probably* not a big
> deal, but I do not know what OpenSSL does when you tell it the random
> engine doesn't support RAND_bytes as opposed to telling it that it's broken.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2073/files#r33426295>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084